Lib/multiprocessing/pool.py

```diff
@@ -570,7 +581,9 @@ def _terminate_pool(cls, taskqueue, inqueue, outqueue, pool,
         util.debug('helping task handler/workers to finish')
         cls._help_stuff_finish(inqueue, task_handler, len(pool))
 
-        assert result_handler.is_alive() or len(cache) == 0
+        if (not result_handler.is_alive()) and (len(cache) != 0):
+            raise AssertionError(
+                "Cannot have cache with result_hander not alive")
```

Comment:
I'm not sure I get the point of adding a message here. The message will not be understandable for the average user, and whoever wants to debug it has to read the source code anyway.

Reply:
How does one tell where the AssertionError is happening, if the pool is being used not by the primary process but by a subprocess (possibly on another machine via a manager)?

Reply:
Thanks to the line number in the traceback, I guess?

Reply:
I wish that worked in the circumstances I noted above - see [here](https://travis-ci.org/drallensmith/neat-python/jobs/262839337), noting that it's a "nightly" build so includes the earlier patch, for an example.